.gitignore

```diff
@@ -1 +1,3 @@
 *~
+.history/
+math/.cpcache/
```

Comment:
We've been using separate `.gitignore` files in each of the subdirectories, so I think we should stick with that pattern unless we have a broader conversation about switching to a single/main `.gitignore`.